corehq/apps/users/views/__init__.py

```diff
@@ -725,15 +724,19 @@ def __call__(self, request, invitation_id, **kwargs):
             logout(request)
             return HttpResponseRedirect(request.path)
 
-        try:
-            invitation = SQLInvitation.objects.get(id=int(invitation_id))
-        except (ValueError, SQLInvitation.DoesNotExist):
-            invitation = SQLInvitation.objects.filter(couch_id=invitation_id).first()
-            if not invitation:
-                messages.error(request, _("Sorry, it looks like your invitation has expired. "
-                                          "Please check the invitation link you received and try again, or "
-                                          "request a project administrator to send you the invitation again."))
-                return HttpResponseRedirect(reverse("login"))
+        invitation = SQLInvitation.objects.filter(uuid=uuid).first()
+        if not invitation:
+            try:
+                invitation = SQLInvitation.objects.get(id=int(uuid))
+            except (ValueError, SQLInvitation.DoesNotExist):
+                invitation = SQLInvitation.objects.filter(couch_id=uuid).first()
```

Comment:
Why we already have `id` and `couch_id`?

Reply:
Right now there are two kinds of invitations in the wild: old ones where the URL references and old couch id and new ones where the URL references a sql id. This PR will add a third kind, with URLs that reference the new uuid. So this code has to handle all three. https://github.com/dimagi/commcare-hq/pull/26687 cleans up this code.

Reply:
woah, okay thanks.